pkg/apis/core/validation/validation.go

```diff
@@ -5360,9 +5360,33 @@ func ValidateSecurityContext(sc *core.SecurityContext, fldPath *field.Path) fiel
 // maxGMSACredentialSpecLength is the max length, in bytes, for the actual contents
 // of a GMSA cred spec. In general, those shouldn't be more than a few hundred bytes,
 // so we want to give plenty of room here while still providing an upper bound.
+// The runAsUserName field will be used to execute the given container's entrypoint, and
+// it can be formatted as "DOMAIN/USER", where the DOMAIN is optional, maxRunAsUserNameDomainLength
+// is the max character length for the user's DOMAIN, and maxRunAsUserNameUserLength
+// is the max character length for the USER itself. Both the DOMAIN and USER have their
+// own restrictions, and more information about them can be found here:
+// https://support.microsoft.com/en-us/help/909264/naming-conventions-in-active-directory-for-computers-domains-sites-and
+//  https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.localaccounts/new-localuser?view=powershell-5.1
 const (
 	maxGMSACredentialSpecLengthInKiB = 64
 	maxGMSACredentialSpecLength      = maxGMSACredentialSpecLengthInKiB * 1024
+	maxRunAsUserNameDomainLength     = 256
+	maxRunAsUserNameUserLength       = 21
+)
+
+var (
+	// a valid NetBios Domain name cannot start with a dot, has at least 1 character,
+	// at most 15 characters, and it cannot the characters: \ / : * ? " < > |
+	// we're also excluding the control characters (ASCII codes: 0-31, 127; in hexa: 00-1F, 7F)
+	validNetBiosRegex, _ = regexp.Compile(`^(?:[^\x00-\x1F\x7F\\/:\*\?"<>|\.][^\x00-\x1F\x7F\\/:\*\?"<>|]{0,14})?$`)
+
+	// a valid DNS name contains only alphanumeric characters, dots, and dashes.
+	validDNSRegex, _ = regexp.Compile(`^(?:(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z0-9][a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?$`)
```

Comment:
the final `?` allows validDNSRegex to match empty strings, which doesn't seem correct

Reply:
`[a-zA-Z0-9][a-zA-Z0-9-]{0,61}[a-zA-Z0-9]` requires the final segment to be at least two characters long... is that correct?

Reply:
the leading pattern of 
```
  (?:
    [a-zA-Z0-9]
    (?:
      [a-zA-Z0-9-]{0,61}
      [a-zA-Z0-9]
    )?
    \.
  )+
```
means that only multi-label DNS names are allowed (e.g. `local-computer-name` would not be allowed). is that correct?

Reply:
> [a-zA-Z0-9][a-zA-Z0-9-]{0,61}[a-zA-Z0-9] requires the final segment to be at least two characters long... is that correct?

that question is still pending. if a single character final segment is allowed, we need a test exercising it (e.g. `some-dns-domain-name.a\\User`)